public/pages/DefineDetector/components/DataFilterList/components/__tests__/DataFilter.test.tsx

```diff
@@ -0,0 +1,194 @@
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { DATA_TYPES } from '../../../../../../utils/constants';
+import { CoreServicesContext } from '../../../../../../components/CoreServices/CoreServices';
+import {
+  coreServicesMock,
+  httpClientMock,
+} from '../../../../../../../test/mocks';
+import { DataFilter } from '../DataFilter';
+import {
+  OPERATORS_MAP,
+  FILTER_TYPES,
+  UIFilter,
+} from '../../../../../../models/interfaces';
+import { DetectorDefinitionFormikValues } from '../../../../models/interfaces';
+import { FormikProps, Formik } from 'formik';
+import { Provider } from 'react-redux';
+import configureStore, { MockStore } from 'redux-mock-store';
+import clientMiddleware from '../../../../../../redux/middleware/clientMiddleware';
+import { AppState } from '../../../../../../reducers';
+
+const initialState = {
+  opensearch: {
+    indices: [
+      {
+        label: 'test-index',
+        health: 'green',
+      },
+    ],
+    aliases: [],
+    dataTypes: {
+      integer: ['cpu', 'memory'],
+    },
+    requesting: false,
+    searchResult: {},
+    errorMessage: '',
+  },
+};
+
+const filters = [
+  {
+    filterType: FILTER_TYPES.SIMPLE,
+    fieldInfo: [
+      {
+        label: 'cpu',
+        type: DATA_TYPES.NUMBER,
+      },
+    ],
+    operator: OPERATORS_MAP.IS_GREATER,
+    fieldValue: 0,
+  },
+] as UIFilter;
+
+const values = {
+  name: 'test-ad',
+  description: 'desc',
+  index: [
+    {
+      label: 'test-index',
+      health: 'green',
+    },
+  ],
+  filters: [
+    {
+      filterType: FILTER_TYPES.SIMPLE,
+      fieldInfo: [
+        {
+          label: 'cpu',
+          type: DATA_TYPES.NUMBER,
+        },
+      ],
+      operator: OPERATORS_MAP.IS_GREATER,
+      fieldValue: 0,
+    },
+  ],
+  filterQuery: JSON.stringify({ bool: { filter: [] } }, null, 4),
+  timeField: 'timestamp',
+  interval: 10,
+  windowDelay: 1,
+} as DetectorDefinitionFormikValues;
+
+const formikProps = {
+  values: { values },
+  errors: {},
+  touched: {
+    index: true,
+    name: true,
+    timeField: true,
+  },
+  isSubmitting: false,
+  isValidating: false,
+  submitCount: 0,
+  initialErrors: {},
+  initialTouched: {},
+  isValid: true,
+  dirty: true,
+  validateOnBlur: true,
+  validateOnChange: true,
+  validateOnMount: true,
+} as FormikProps<DetectorDefinitionFormikValues>;
+
+const mockedStore = (mockState = initialState): MockStore<AppState> => {
+  const middlewares = [clientMiddleware(httpClientMock)];
+  const mockStore = configureStore<AppState>(middlewares);
+  const store = mockStore(mockState);
+  return store;
+};
+describe('dataFilter', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  test('renders data filter', async () => {
+    const { container, getByText, getByTestId, getAllByText } = render(
```

Comment:
the `render` fn here seems identical to `render` on lines 158-181; can you pull those out into a general fn? Ref: https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/blob/main/public/pages/DetectorsList/containers/List/__tests__/List.test.tsx#L34

Reply:
Refactored it